pkg/validation/attributes/attributes.go

```diff
@@ -0,0 +1,62 @@
+package attributes
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	apiAttributes "github.com/devfile/api/v2/pkg/attributes"
+)
+
+// ValidateGlobalAttribute validates the workspace template spec data for global attribute references
+func ValidateGlobalAttribute(workspaceTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec) error {
+
+	var err error
+
+	if workspaceTemplateSpec != nil {
+		// Validate the components
+		if err = ValidateComponents(workspaceTemplateSpec.Attributes, &workspaceTemplateSpec.Components); err != nil {
+			return err
+		}
+
+		// Validate the commands
+		if err = ValidateCommands(workspaceTemplateSpec.Attributes, &workspaceTemplateSpec.Commands); err != nil {
+			return err
+		}
+
+		// Validate the events
+		if err = ValidateEvents(workspaceTemplateSpec.Attributes, workspaceTemplateSpec.Events); err != nil {
+			return err
+		}
+
+		// Validate the projects
+		if err = ValidateProjects(workspaceTemplateSpec.Attributes, &workspaceTemplateSpec.Projects); err != nil {
+			return err
+		}
+
+		// Validate the starter projects
+		if err = ValidateStarterProjects(workspaceTemplateSpec.Attributes, &workspaceTemplateSpec.StarterProjects); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var globalAttributeRegex = regexp.MustCompile(`\{{2}(.*?)\}{2}`)
```

Comment:
Nitpick (feel free to ignore): I feel like using the repetition regexp obscures the meaning here since the char being checked is `{`. Perhaps `\{\{(.*?)\}\}` is clearer? We may also want to exclude `{` and `}` from the inner match, to avoid weirdness around using e.g. `{{my{attribute}}`, but that's probably disallowed elsewhere.

Reply:
`my{attribute` is a valid string for an attribute key, if we exclude them, are we restricting it too much? 🤔

Reply:
We're fully in the weeds of edge cases now :)

How would we parse `{{my{{attribute}}test}}`?

Reply:
The first occurrence of `{{` & `}}` ie, `my{{attribute` is the match and remaining `test}}` is considered a string, to keep it fairly simply and consistent 🤔